input.go

```diff
@@ -92,6 +92,29 @@ func IsMouseButtonPressed(mouseButton MouseButton) bool {
 	return uiDriver().Input().IsMouseButtonPressed(driver.MouseButton(mouseButton))
 }
 
+// GamepadGUID returns a string with the uuid.
+//
+// GamepadGUID is concurrent-safe.
+//
+// GamepadGUID always returns an empty string on browsers and mobiles.
+func GamepadGUID(id int) string {
+	return uiDriver().Input().GamepadGUID(id)
+}
+
+// GamepadName returns a string with the name.
+// This function may vary in how it returns descriptions for the same device across platforms
+// for example the following drivers/platforms see a Xbox One controller as the following:
+// - Windows: "Xbox Controller"
+// - Chrome: "Xbox 360 Controller (XInput STANDARD GAMEPAD)"
+// - Firefox: "xinput"
```

Comment:
In the first place, what is the purpose to get the name of the game pad? It looks like the name is not portable, then this looks not useful to detect the gamepad type.

Reply:
I want the in-game instructions with game controls to display A,B,X,Y if we detect that its a Xbox or Xbox 360 Controller. Likewise if we detect that a PS4 controller is in, then we want the controls to reflect that too.

If we don't fuzzily detect either of those controller types, then we fallback to displaying buttons as the less intuitive "Button 0", "Button 1", etc.

Reply:
As there are various possibilities of the names, I was wondering how you could detect the gamepad is Xbox's or not.

Reply:
I've put a code snippet of what I'm going to do for now. I'm not claiming this would be the most robust / error-free way to do it but I have no better ideas. I'm yet to test with a PS3/PS4 controller so I haven't added that logic in yet.

Worth noting that I also check for what input was last pressed (keyboard/gamepad) and I swap the buttons shown on-screen accordingly.

Finally, here's a resource I'll be referring to in the future to make this check cover more cases:
https://github.com/sriharshachilakapati/GLFW-Controller-Database

```
const deviceId = 0

type GamepadShape int32

const (
	GamepadShapeStandard    GamepadShape = 0
	GamepadShapeXbox        GamepadShape = 1
	GamepadShapePlaystation GamepadShape = 2
)

func updateLoop() {
	// Detect controller
	gamepadShape = GamepadShapeStandard
	description := gml.GamepadGetDescription(deviceId)
	if description != "" {
		if strings.HasPrefix(description, "Xbox Controller") ||
			strings.HasPrefix(description, "Xbox 360 Controller") ||
			strings.HasPrefix(description, "xinput") {
			gamepadShape = GamepadShapeXbox
		}
	}
}
```

Reply:
I was wondering if there is a good list including browsers.

Reply:
If there is, I'd want to add a link in the comment. What do you think?

Reply:
ping

Reply:
I'm not aware of anything comprehensive!
I just inlined what I saw with some of my manual testing with the controllers I have.

Sorry for the delayed reply :)

Reply:
Acknowledged.